### DISCUSSION TITLE
Object reference to the state in a component  .. we are able to change the zustand state without using setHook

### DISCUSSION THREAD

**Discussion by web4-var** [pmndrs/zustand#1471] [Q&A]
Hi All, 

I have a shell container which stores some user info in the zustand store (a vanilla store) 

const vanillaStore = createVanilla<GlobalState>()(
  devtools((set) => ({
    userInfo: {
     name:'nancy',
     postId: '23'
    },

export const { getState, subscribe } = vanillaStore;
export const useGlobalStore = create(vanillaStore);

to display the name in dashboard i am doing
const { userInfo } = useStore((state) => ({
    ...state,
  }));

In this dashboardComponent i can change the globalState just by referencing the userInfo and adding userInfo.name ='Smith' and the globalStore gets updated with this data ? Is this a valid scenario where any component which has access to store can just update it with object reference 

The thing is if i expose this Store to a subapp (webpack 5 module federation) a subapp can just reference this and update the store with object reference. Is there a way we can avoid any subapp to not update the state with zustand but just consume the state with the get hook

**Comment by dai-shi** [comment]
The state object must be immutable by contract.
If you need to somehow force it, you could deep-freeze the object. https://www.npmjs.com/search?q=deep%20freeze

```js
const { userInfo } = deepFreeze(useGlobalStore())
```

**Reply by web4-var** [reply]
hi @dai-shi , 
when you say the state object must be immutable by contract .. how does one achieve this

**Reply by dai-shi** [reply]
I meant we assume so, and follow as a rule. You can restrict so with `freeze` if you want.
